src/ts/signers.ts

```diff
@@ -16,13 +16,16 @@ import {
  * Takes a Signer instance on creation.
  * All other Signer methods are proxied to initial instance.
  */
-export class TypedDataV3Signer implements TypedDataSigner {
+export class TypedDataVersionedSigner implements TypedDataSigner {
```

Comment:
nits: To make this a non breaking change, do you think we should add:
```
export class TypedDataV3Signer extends TypedDataVersionedSigner {
  constructor(signer: Signer) {
    super(signer, "v3");
  }
}
```

Reply:
Happy to do it, its just that I think we were the only ones using it so i preferred not to add noise in the code. 
But you are technically right that becomes a breaking change, so happy to do it.  If you are inclined to add it, let me know.

Reply:
I do think we should do it. While I don't think it is widely used, I do think we should respect semver.

If a lot of extra code would be required, I would agree with you, but since its just a small bit of boilerplate, I think its better in this case to keep the backwards compatible interface.